Frontend: Add Service (as Admin) + View Services (everyone)

sandyl289 opened the pull request:
Branch: `frontend/add-service`

- All tests pass (Repository & Service) _I ran the tests just in case_
- `Add Service` (as Admin) (https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/169)
- `View Services` (everyone) (https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/170)
- Change service param to ServiceDto
- Modified navbar to only let Admin see "Add Service"

Thanks Tyler for helping me! 

# Add New Service (Must be logged in as Admin)

## Add New Service
![image](https://user-images.githubusercontent.com/70775435/112545540-b666e080-8d8e-11eb-84bd-fbff3d12846a.png)

## Successful Message
![image](https://user-images.githubusercontent.com/70775435/112545582-c252a280-8d8e-11eb-8c0f-1dd595e6a74a.png)

## If service name already exists
![image](https://user-images.githubusercontent.com/70775435/112545686-e44c2500-8d8e-11eb-8cc3-f2bc8ca74bd1.png)

# View Services

## If no service yet
![image](https://user-images.githubusercontent.com/70775435/112546275-a0a5eb00-8d8f-11eb-80b4-1fb903e9acce.png)

## If there are services
![image](https://user-images.githubusercontent.com/70775435/112546520-e498f000-8d8f-11eb-848a-23ac800f7559.png)

JDziewonski98 committed:
started login and navbar

JDziewonski98 committed:
added register tab
store logged in user state
conditional rendering on register

sandyl289 committed:
Merge branch 'main' into frontend/add-service

JDziewonski98 committed:
added logout and tweaked other pages

JDziewonski98 committed:
Merge branch 'main' into frontend-login

JDziewonski98 committed:
added 'logged in as'

JDziewonski98 committed:
Merge branch 'main' into frontend-login

sandyl289 committed:
Static components (Add Service & View Service)

sandyl289 committed:
Trying dynamic components

sandyl289 committed:
Merge branch 'frontend-login' into frontend/add-service

sandyl289 committed:
Fix View Service (by Tyler)

sandyl289 committed:
Fix Add Service & View Service

sandyl289 committed:
Fix error message for existing service (by Tyler)

sandyl289 committed:
Reset successful message  + Remove useless comments

sandyl289 committed:
Fix adding same service name with space (Thanks to Tyler)

- All tests pass (Repository & Service)

T-Syme commented on .vscode/launch.json:
```diff
@@ -0,0 +1,15 @@
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
Interesting that this file isn't included in the .gitignore... It shouldn't matter though

JDziewonski98 commented on RepairShop-Frontend/src/components/AddService.vue:
```diff
@@ -0,0 +1,133 @@
+  //PRODUCTION
+  //baseURL: "http://" + config.build.backendHost + ":" + config.build.backendPort,
+
+  headers: {
```
Does the call work without this?

JDziewonski98 commented on RepairShop-Frontend/src/components/Logout.vue:
```diff
@@ -1,7 +1,7 @@
   <div id="titleContainer">
     <h1 v-show="success">Logged Out Successfully!</h1>
-    <h1 v-show="!success">Your were not logged in.</h1>
+    <h1 v-show="!success">You were not logged in.</h1>
```
hmm why is this stuff showing up? I though I already changes this on main

JDziewonski98 commented on RepairShop-Frontend/src/components/AddService.vue:
```diff
@@ -0,0 +1,133 @@
+
+    <p>
+      <span v-if="successAddService" style="color: green">
+        {{ successAddService }}
```
I find this a bit hard to notice, maybe make it a bolder color or make it flash? Not a big deal at all

JDziewonski98 reviewed (approved):
Looks good to me, nice work

sandyl289 committed:
Remove useless comments

sandyl289 committed:
Make successful message dark green

- All tests pass (Repository & Service)

sandyl289 committed:
Input validation for duration & price

- Change message of no services to show (ViewService.vue)
- Let user to only enter positive numbers (no string)
- All tests pass (Repository & Service)

sandyl289 committed:
Merge branch 'main' into frontend/add-service

- All tests pass (Repository & Service)

sandyl289 commented:
I'll close https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/169 and https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/170
